src/virt_card.py

```diff
@@ -97,13 +100,22 @@ def run_cmd(self, cmd: str = None, expect: str = None, pin: bool = True, passwd:
                 shell.sendline(passwd)
 
             if expect is not None:
-                out = shell.expect([pexpect.TIMEOUT, expect], timeout=20)
-
-                if out != 1:
-                    if out == 0:
-                        log.error("Time out")
-                    raise pexpect.exceptions.EOF(f"Pattern '{expect}' is not "
-                                                 f"found in the output.")
+                if reject is not None:
+                    out = shell.expect([reject, pexpect.TIMEOUT, expect], timeout=20)
+                    if out != 2:
+                        if out == 1:
+                            log.error("Time out")
+                        if out == 0;
+                            log.error("Disallowed pattern found")
+                        raise pexpect.exceptions.EOF(f"Pattern '{reject}' is "
```

Comment:
This exception would be raised also when reject pattern is not found, but it is a Time out. Message here corresponds only to the case if pattern is realy found in the output, because imaging situation when mathced pattern pattern is not 2, **reject pattern is not in the output**, but exception would say "Pattern .. is found". So I would suggest to move this exception under if block

Reply:
Thanks for the comment, I agree. I will slightly rework it in next commit.